### ISSUE TITLE
Blocage de la Freebox Player

### ISSUE THREAD

**Issue by Novicami** [betonniere/freeteuse#3]
J'utilise la Freeteuse depuis peu de temps car ma télécommande ne fonctionne plus correctement. J'ai remarqué à plusieurs reprises que le changement de volume peut occasionnellement un blocage du Player. Le volume monte (ou descend) rapidement au maximum, la barre de volume reste constamment affichée et la télécommande ne répond plus. En attendant, je vais changer de télécommande virtuel pour voir si cela est un problème récurrent.

**Comment by betonniere** [comment]
Le problème m'arrive également de temps en temps.

Pour info la répétition des touches est gérée non pas par l'appli, mais par la freebox.
Avec le SDK de Free, les applis ont pour tâche de transmette uniquement les évènements : "**touche appuyée**" et "**touche relachée**". A priori c'est ce que fait l'appli.
C'est pas encore une certitude mais je me demande si on ne serait pas face à un bug du FreePlayer qu'il faudrait que j'essaye de contourner.

**Comment by Novicami** [comment]
Pourquoi pas simuler des impulsions temporisées?

**Comment by betonniere** [comment]
Oui, tout à fait. C'est à ce que je pensais comme contournement.

**Comment by betonniere** [comment]
Je viens de passer en revu des commentaires d'autres télécommandes Freebox et plusieurs utilisateurs remontent le même problème. C'est donc très probablement un bug de la Freebox.
Du coup je vais tâcher de mettre en place le contournement.

**Comment by Novicami** [comment]
Je ne sais pas si c'est à prendre en compte pour l'appli, mais lorsque l'on utilise une vrai télécommande, on constate sur la box qu'un appui long se traduit par des impulsions, ce qui peut donner le bon tempo si tu arrive à le simuler!

**Comment by Novicami** [comment]
Je constate la disparition du bogue en question en ce moment, c'est lié aux dernière mise à jours?
Sinon question bête sans doute mais que ce passe-t-il si lors d'un appui long (ex:volume), la connexion WIFI rencontre des micro coupures? Genre l'évènement "touche relachée" n'est pas transmise, c'est possible?

**Comment by betonniere** [comment]
J'ai également cru que le problème avait disparu, mais je suis tombé dessus 1 fois ce week-end. En tous cas il est beaucoup plus rare de mon point de vue et il est bien possible que Free ait fait une mise de son firmware expliquant cette amélioration.

Tu as probablement raison en évoquant les micros coupures WIFI. Le protocole sur lequel se repose le SDK est de l'UDP et non pas du TCP. UDP est véloce certe, mais non fiable par nature. Pour une raison ou pour une autre les messages n'ont pas la garantie d'être acheminés et peuvent être perdus. Le SDK de Free n'a pas été conçu à prioris avec cette contrainte en tête et il y a de grandes chances que ce soit la source de notre problème. En tous cas il s'explique très bien si le message "touche relachée" est perdu.

J'ai fait des essais de mon côté en gérant les appuis longs à partir de l'appli par des impulsions temporisées. La fuidité n'est malheureusement pas du tout la même et risque de paraître plutôt pénible pour les utilisateurs. Du coup j'ai abandonné cette piste comme contournement.
